spring-pulsar-test/src/main/java/org/springframework/pulsar/test/support/PulsarTestConsumer.java

```diff
@@ -0,0 +1,72 @@
+package org.springframework.pulsar.test.support;
+
+import static org.apache.pulsar.client.api.SubscriptionInitialPosition.Earliest;
+import static org.apache.pulsar.client.api.SubscriptionInitialPosition.Latest;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.pulsar.client.api.Consumer;
+import org.apache.pulsar.client.api.Message;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.api.SubscriptionInitialPosition;
+
+import org.springframework.pulsar.core.DefaultPulsarConsumerFactory;
+
+public class PulsarTestConsumer {
+
+	private final DefaultPulsarConsumerFactory<?> consumerFactory;
+
+	private final Duration timeout;
+
+	public PulsarTestConsumer(DefaultPulsarConsumerFactory<?> consumerFactory, Duration timeout) {
+		this.consumerFactory = consumerFactory;
+		this.timeout = timeout;
+	}
+
+	public PulsarTestConsumer(DefaultPulsarConsumerFactory<?> consumerFactory) {
+		this(consumerFactory, Duration.ofSeconds(10));
+	}
+
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	private <T> List<Message<T>> receiveEarliestMessages(int desiredMessageCount, List<String> topics, Schema<T> schema,
+			SubscriptionInitialPosition initialPosition, String subscriptionName) {
+		try (Consumer<T> consumer = consumerFactory.createConsumer((Schema) schema, topics, subscriptionName,
+				c -> c.subscriptionInitialPosition(initialPosition))) {
+			long remainingMillis = timeout.toMillis();
```

Comment:
I did not get too deep into the impl but since this will be consumed by tests, we can use Awaitility to do this for us very easily.

Reply:
Indeed I was thinking the same while implementing it. Think I mimicked it too much after `KafkaTestUtils`